include/seqan3/range/decorator/gap_decorator_anchor_set.hpp

```diff
@@ -67,10 +66,16 @@ namespace seqan3
  * Resolving random access requires logarithmic access into the set and inserting or removing a gap symbol additionally
  * entails updating all subsequent elements in the set to preserve correct cumulative sizes.
  *
+ * ### The seqan3::gap_decorator_anchor_set::iterator type
+ *
+ * \attention The iterator of the seqan3::gap_decorator_anchor_set does not model the std::LegacyInputIterator because
+ * it has no operator->. Note that it does model the std::ranges::InputIterator.
+ *
  */
-template <std::ranges::RandomAccessRange inner_type>
+template <std::ranges::ViewableRange inner_type>
```

Comment:
I would propose constraining this to std::ranges::View and use two constructors. The first takes this type as a copy and moves into the respective data member. The other is a ref_view wrapped non-view range deduced by the deduction guide. Then you need only one. Or was there a specific reason not to do it this way?